coil-gif/src/main/java/coil3/gif/AnimatedImageDecoder.kt

```diff
@@ -122,7 +124,9 @@ class AnimatedImageDecoder(
             return baseDrawable
         }
 
-        baseDrawable.repeatCount = options.repeatCount
+        if (options.repeatCount != ENCODED_LOOP_COUNT) {
```

Comment:
> We'll need to add handling to `GifDecoder.Factory` to ensure we don't set the repeat count there either when `options.repeatCount == ENCODED_LOOP_COUNT`.

I'm trying to add check() to the Factory as shown below. What do you think about this?

```kotlin
class Factory(
    val enforceMinimumFrameDelay: Boolean = true,
) : Decoder.Factory {

    override fun create(
        result: SourceFetchResult,
        options: Options,
        imageLoader: ImageLoader,
    ): Decoder? {
        if (Build.VERSION.SDK_INT >= 28) {
            check(options.repeatCount == ENCODED_LOOP_COUNT) { 
                "ENCODED_LOOP_COUNT cannot be used with GifDecoder. Please use the AnimatedImageDecoder if you need to use the encoded loop count."
            }
        }

        if (!DecodeUtils.isGif(result.source.source())) return null
        return GifDecoder(result.source, options, enforceMinimumFrameDelay)
    }
}
```

Reply:
We only want to guard against it here and ignore the parameter instead of throwing: https://github.com/coil-kt/coil/blob/main/coil-gif/src/main/java/coil3/gif/GifDecoder.kt#L52

Reply:
Is this also possible in the following way?

```kotlin
        if (options.repeatCount != ENCODED_LOOP_COUNT) {
            drawable.setRepeatCount(options.repeatCount)
        }
```